src/lxc/conf.c

```diff
@@ -2741,6 +2741,16 @@ static int instantiate_vlan(struct lxc_handler *handler, struct lxc_netdev *netd
 
 	DEBUG("instantiated vlan '%s', ifindex is '%d'", " vlan1000",
 	      netdev->ifindex);
+	if (netdev->mtu) {
+		int mtu = atoi(netdev->mtu);
+		err = mtu > 64 && mtu < 65536 ? lxc_netdev_set_mtu(peer,mtu) : EINVAL;
+		if(err) {
+			ERROR("failed to set mtu '%s' for %s : %s",
+			      netdev->mtu, peer, strerror(-err));
```

Comment:
lxc_netdev_set_mtu() return code < 0 on error. strerror(-err) - it's right.
I removed the extra checking mtu. I hope that the kernel will carefully examine this option.  Maybe this check need to be moved to lxc_netdev_set_mtu () function?